PantryNodeReact/src/pages/login.tsx

```diff
@@ -13,21 +13,76 @@ import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
 import Typography from '@mui/material/Typography';
 import Container from '@mui/material/Container';
 import Copyright from '../Components/Copyright';
-
+import { useState } from "react";
 
 
 export default function Login() {
+
+  // validation
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [emailError, setEmailError] = useState("");
+  const [passwordError, setPasswordError] = useState("");
+  const [remember, setRemember] = useState(false);
+
+  const handleRememberChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setRemember(event.target.checked);
+  };
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+    if (name === "email") {
+      setEmail(value);
+      setEmailError("");
+    } else if (name === "password") {
+      setPassword(value);
+      setPasswordError("");
+    }
+  };
+
+
   const navigate = useNavigate();
   const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault();
     const data = new FormData(event.currentTarget);
+
+    const emailValue = data.get("email") as string;
+    const passwordValue = data.get("password") as string;
+
+    // email
+    if (emailValue.trim() === "") {
+      setEmailError("Email is required");
+      return;
+    }
+  
+    if (!/\S+@\S+\.\S+/.test(emailValue)) {
+      setEmailError("Email is invalid");
+      return;
+    }
+    
+    // password
+    if (passwordValue.trim() === "") {
+      setPasswordError("Password is required");
+      return;
+    }
+    if (passwordValue.length < 8){
+      setPasswordError("Password must be at least 8 characters long");
+      return;
+    }
+    if (!/(?=.*[A-Z])(?=.*[\W_])/.test(passwordValue)) {
+      setPasswordError("Password must contain at least one number and one special character");
+      return;
+    }
+
     console.log({
       email: data.get('email'),
       password: data.get('password'),
     });
```

Comment:
do we need to log user data ?

Reply:
Not really part of this PR but originally, I suppose this was for logging purpose to check what data to pass on other end points, but can remove this though, what do you say @prasannarajezzzy

Reply:
Cool